src/databricks/labs/remorph/reconcile/query_builder/threshold_query.py

```diff
@@ -30,3 +35,61 @@ def build_query(self) -> str:
     def _construct_threshold_query(table, query_filter, col_expr) -> str:
         expr = ",".join(col_expr)
         return f"select {expr} from {table} where {query_filter}"
+
+    # Comparison query
+    def build_comparison_query(self) -> str:
+        threshold_columns = sorted(set(self.table_conf.get_threshold_columns))
+        join_columns = sorted(set(self.table_conf.get_join_columns))
+        # generate select clause and where clause
+        select_clause, where_clause = self._generate_select_where_clause(threshold_columns)
+        # add join columns to select clause
+        select_clause.extend([f"source.{column} as {column}" for column in join_columns])
+        # generate join condition dynamically
+        join_condition = self._get_join_condition(join_columns)
+        # construct the final query
+        comparison_query = self._construct_comparison_query(
+            self.table_conf, select_clause, join_condition, where_clause
+        )
+
+        logger.info(f"Comparison Query: {comparison_query}")
+        print(f"Comparison Query: {comparison_query}")
+        return comparison_query
+
+    def _generate_select_where_clause(self, threshold_columns: set[str]) -> (list[str], list[str]):
+        select_clause = []
+        where_clause = []
+        for column in threshold_columns:
+            # get the user provided threshold details at column level
+            threshold = self._get_threshold_info(column)
+            # check if the threshold is in percentage or absolute mode
+            mode = (
```

Comment:
Could the logic around the `%` sign, ie. picking the right `ThresholdMode` and normalizing the values of `lower_bound` and `upper_bound`, be moved to the `Thresholds` class?

Reply:
Done. Moved to dataclass